test/framework/systemtools.py

```diff
@@ -387,7 +387,7 @@ def test_os_version(self):
     def test_gcc_version_native(self):
         """Test getting gcc version."""
         gcc_version = get_gcc_version()
-        self.assertTrue(isinstance(gcc_version, basestring) or gcc_version == UNKNOWN)
+        self.assertTrue(isinstance(gcc_version, basestring) or gcc_version == UNKNOWN or gcc_version is None)
```

Comment:
There's no (real) GCC on OS X, then it's `None` (see also the tests below).

If `gcc --version` fails to run (exit code != 0), it's set to unknown.